src/Middleware/OutputCaching/test/OutputCacheMiddlewareTests.cs

```diff
@@ -826,35 +827,86 @@ public async Task Locking_PreventsConcurrentRequests()
         Assert.Equal(1, responseCounter);
     }
 
+    [Fact]
+    public async Task Locking_IgnoresNonCacheableResponses()
+    {
+        var responseCounter = 0;
+
+        var blocker = new TaskCompletionSource<bool>();
+
+        var memoryStream1 = new MemoryStream();
+        var memoryStream2 = new MemoryStream();
+
+        var options = new OutputCacheOptions();
+        options.AddBasePolicy(build => build.Cache());
+
+        var middleware = TestUtils.CreateTestMiddleware(options: options, next: async c =>
+        {
+            responseCounter++;
+
+            if (responseCounter == 1)
+            {
+                blocker.SetResult(true);
+                await Task.Delay(500);
```

Comment:
😢 any way to use a synchronization primitive instead? Or is the second request waiting purely internal and not observable in any way?

Reply:
There is no code path to unblock the second request at a known moment.